guides/keras_cv/object_detection_keras_cv.py

```diff
@@ -0,0 +1,872 @@
+    640, 640, bounding_box_format="xywh", pad_to_aspect_ratio=True
+)
+eval_ds = eval_ds.map(inference_resizing, num_parallel_calls=tf.data.AUTOTUNE)
+
+"""
+Due to the fact that the resize operation differs between the train dataset,
+which uses `JitteredResize()` to resize images, and the inference dataset, which
+uses `layers.Resizing(pad_to_aspect_ratio=True)`. it is good practice to
+visualize both datasets:
+"""
+
+visualize_dataset(
+    eval_ds, bounding_box_format="xywh", value_range=(0, 255), rows=2, cols=2
+)
+
+"""
+Finally, let's unpackage our inputs from the preprocessing dictionary, and
+prepare to feed the inputs into our model.  In order to be TPU compatible,
+bounding box Tensors need to be `Dense` instead of `Ragged`.  If training on
+GPU, you can omit the `bounding_box.to_dense()` call.  If ommitted,
+the KerasCV RetinaNet
+label encoder will automatically correctly encode Ragged training targets.
+"""
+
+
+def dict_to_tuple(inputs):
+    return inputs["images"], bounding_box.to_dense(
+        inputs["bounding_boxes"], max_boxes=32
+    )
+
+
+train_ds = train_ds.map(dict_to_tuple, num_parallel_calls=tf.data.AUTOTUNE)
+eval_ds = eval_ds.map(dict_to_tuple, num_parallel_calls=tf.data.AUTOTUNE)
+
+train_ds = train_ds.prefetch(tf.data.AUTOTUNE)
+eval_ds = eval_ds.prefetch(tf.data.AUTOTUNE)
+
+"""
+
+### Optimizer
+
+In this guide, we use a standard SGD optimizer and rely on the
+[`keras.callbacks.ReduceLROnPlateau`](https://keras.io/api/callbacks/reduce_lr_on_plateau/)
+callback to reduce the learning rate.
+
+You will always want to include a `global_clipnorm` when training object
+detection models.  This is to remedy exploding gradient problems that frequently
+occur when training object detection models.
+"""
+
+base_lr = 0.005
+# including a global_clipnorm is extremely important in object detection tasks
+optimizer = tf.keras.optimizers.SGD(
+    learning_rate=base_lr, momentum=0.9, global_clipnorm=10.0
+)
+
+"""
+To achieve the best results on your dataset, you'll likely want to hand craft a
+`PiecewiseConstantDecay` learning rate schedule.
+While `PiecewiseConstantDecay` schedules tend to perform better, they don't
+translate between problems.
+"""
+
+"""
+### Loss functions
+
+You may not be familiar with the `"focal"` or `"smoothl1"` losses.  While not
+common in other models, these losses are more or less staples in the object
+detection world.
+
+In short, ["Focal Loss"](https://arxiv.org/abs/1708.02002) places extra emphasis
+on difficult training examples.  This is useful when training the classification
+loss, as the majority of the losses are assigned to the background class.
+
+"SmoothL1 Loss" is used to [prevent exploding gradients](https://arxiv.org/abs/1504.08083)
+that often occur when attempting to perform the box regression task.
+
+In KerasCV you can use these losses simply by passing the strings `"focal"` and
+`"smoothl1"` to `compile()`:
+"""
+
+pretrained_model.compile(
```

Comment:
you are compiling your model twice before training it. Consider moving this "losses" block after the instantiation of the model, where a .compile step is relevant.